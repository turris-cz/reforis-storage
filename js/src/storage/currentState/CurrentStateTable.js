import React from "react";
import { SpinnerElement } from "foris";
import { NOT_PENDING_STORAGE_STATES, STORAGE_STATES } from "../constants";

export default function CurrentStateTable({state, old_device_desc, uuid, raid, storageIsPending}) {
    return (
        <table className="table table-borderless table-hover offset-lg-1 col-lg-10 col-sm-12">
            <tbody>
                {!Object.keys(NOT_PENDING_STORAGE_STATES).includes(state) && (
                    <tr>
                        <th scope="row">{_("State")}</th>
                        <td style={{ display: "flex", flexDirection: "row" }}>
                            {storageIsPending ? (
                                <SpinnerElement small>
                                    {STORAGE_STATES[state]}
                                </SpinnerElement>
                            ) : STORAGE_STATES[state]}
                        </td>
                    </tr>
                )}
                {old_device_desc !== "none"
                    && (
                        <tr>
                            <th scope="row">{_("Device")}</th>
                            <td>{old_device_desc}</td>
                        </tr>
                    )}
                <tr>
                    <th scope="row">{_("UUID")}</th>
                    <td>{uuid}</td>
                </tr>
                <tr>
                    <th scope="row">{_("RAID")}</th>
                    <td>{raid}</td>
                </tr>
            </tbody>
        </table>
    );
}
