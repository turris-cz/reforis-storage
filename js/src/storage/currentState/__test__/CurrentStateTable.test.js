/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import { render } from "foris/testUtils/customTestRender";

import diffSnapshot from "snapshot-diff";
import getStateFixture from "../../__tests__/__fixtures__/getStateFixture";
import CurrentStateTable from "../CurrentStateTable";

describe("<CurrentStateTable />", () => {
    let rerender;
    let asFragment;
    let firstRender;

    beforeEach(() => {
        ({ rerender, asFragment } = render(<CurrentStateTable
            storageIsPending={false}
            {...getStateFixture()}
        />));
        firstRender = asFragment();
    });

    it("Should render.", () => {
        expect(firstRender).toMatchSnapshot();
    });

    it("Should render spinner when storage is pending.", () => {
        rerender(
            <CurrentStateTable
                {...getStateFixture()}
                state="formatting"
                storageIsPending
            />,
        );

        expect(diffSnapshot(firstRender, asFragment())).toMatchSnapshot();
    });

    it("Should hide device if no device selected.", () => {
        rerender(
            <CurrentStateTable
                {...getStateFixture()}
                old_device_desc=""
                storageIsPending={false}
                old_uuid="broken"
            />,
        );
        expect(diffSnapshot(firstRender, asFragment())).toMatchSnapshot();
    });
});
