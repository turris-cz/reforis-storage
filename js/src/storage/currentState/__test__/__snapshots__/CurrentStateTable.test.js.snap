// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrentStateTable /> Should hide device if no device selected. 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -9,13 +9,11 @@
          class=\\"col-md-6 d-flex flex-column align-items-center justify-content-center\\"
        >
          <i
            class=\\"fas fa-hdd fa-4x mb-1\\"
          />
-         <h5>
-           /dev/sdb
-         </h5>
+         <h5 />
        </div>
        <div
          class=\\"col-md-6\\"
        >
          <div
@@ -32,13 +30,11 @@
                    <th
                      scope=\\"row\\"
                    >
                      Device
                    </th>
-                   <td>
-                     /dev/sdb
-                   </td>
+                   <td />
                  </tr>
                  <tr>
                    <th
                      scope=\\"row\\"
                    >"
`;

exports[`<CurrentStateTable /> Should render spinner when storage is pending. 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -30,10 +30,36 @@
                <tbody>
                  <tr>
                    <th
                      scope=\\"row\\"
                    >
+                     State
+                   </th>
+                   <td>
+                     <div
+                       class=\\"d-flex flex-row\\"
+                     >
+                       <div
+                         class=\\"spinner-border spinner-border-sm\\"
+                         role=\\"status\\"
+                       >
+                         <span
+                           class=\\"sr-only\\"
+                         />
+                       </div>
+                       <div
+                         class=\\"spinner-text\\"
+                       >
+                         Â Formatting
+                       </div>
+                     </div>
+                   </td>
+                 </tr>
+                 <tr>
+                   <th
+                     scope=\\"row\\"
+                   >
                      Device
                    </th>
                    <td>
                      /dev/sdb
                    </td>"
`;

exports[`<CurrentStateTable /> Should render. 1`] = `
<DocumentFragment>
  <div
    class="mb-3"
  >
    <div
      class="row no-gutters justify-content-center"
    >
      <div
        class="col-md-6 d-flex flex-column align-items-center justify-content-center"
      >
        <i
          class="fas fa-hdd fa-4x mb-1"
        />
        <h5>
          /dev/sdb
        </h5>
      </div>
      <div
        class="col-md-6"
      >
        <div
          class="card-body"
        >
          <div
            class="table-responsive"
          >
            <table
              class="table table-borderless table-hover"
            >
              <tbody>
                <tr>
                  <th
                    scope="row"
                  >
                    Device
                  </th>
                  <td>
                    /dev/sdb
                  </td>
                </tr>
                <tr>
                  <th
                    scope="row"
                  >
                    UUID
                  </th>
                  <td>
                    1234-1234-1234-1234-1234
                  </td>
                </tr>
                <tr>
                  <th
                    scope="row"
                  >
                    RAID
                  </th>
                  <td>
                    single
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
